catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/validation/ValidateExecutor.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.commands.catalog.validation;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.util.Describable;
+import ddf.catalog.validation.MetacardValidator;
+import ddf.catalog.validation.ValidationException;
+
+public class ValidateExecutor {
+
+    public static List<ValidateReport> execute(List<Metacard> metacards, List<MetacardValidator> validators) {
+        return metacards.stream()
+                .map(metacard -> generateReport(metacard, validators))
+                .collect(Collectors.toList());
+    }
+
+    private static ValidateReport generateReport(Metacard metacard, List<MetacardValidator> validators) {
+
+        ValidateReport report = new ValidateReport(metacard.getTitle());
+
+        for (MetacardValidator validator : validators) {
+            try {
+                validator.validate(metacard);
+            } catch (ValidationException e) {
+                String name = validator.getClass().getName();
+                if (validator instanceof Describable && ((Describable) validator).getId() != null) {
+                    name = ((Describable) validator).getId();
+                }
+                List<String> warnings = e.getWarnings() == null ? new ArrayList<>() : e.getWarnings();
```

Comment:
Should use `Collections.emptyList()` or `ImmutableList.of()` for your empty lists.